wp-seo-main.php

```diff
@@ -300,29 +300,31 @@ function wpseo_init() {
 	$link_watcher->load();
 
 	// Loading Ryte integration.
-	if ( WPSEO_Options::get( 'onpage_indexability' ) ) {
-		$wpseo_onpage = new WPSEO_OnPage();
-		$wpseo_onpage->register_hooks();
-	}
+	$wpseo_onpage = new WPSEO_OnPage();
+	$wpseo_onpage->register_hooks();
 }
 
 /**
  * Loads the rest api endpoints.
  */
 function wpseo_init_rest_api() {
 	// We can't do anything when requirements are not met.
-	if ( WPSEO_Utils::is_api_available() ) {
-		// Boot up REST API.
-		$configuration_service = new WPSEO_Configuration_Service();
-		$configuration_service->initialize();
+	if ( ! WPSEO_Utils::is_api_available() ) {
+		return;
+	}
+
+	// Boot up REST API.
+	$configuration_service = new WPSEO_Configuration_Service();
+	$configuration_service->initialize();
 
-		$link_reindex_endpoint = new WPSEO_Link_Reindex_Post_Endpoint( new WPSEO_Link_Reindex_Post_Service() );
-		$link_reindex_endpoint->register();
+	$link_reindex_endpoint = new WPSEO_Link_Reindex_Post_Endpoint( new WPSEO_Link_Reindex_Post_Service() );
+	$link_reindex_endpoint->register();
 
-		$statistics_service  = new WPSEO_Statistics_Service( new WPSEO_Statistics() );
-		$statistics_endpoint = new WPSEO_Endpoint_Statistics( $statistics_service );
-		$statistics_endpoint->register();
+	$statistics_service  = new WPSEO_Statistics_Service( new WPSEO_Statistics() );
+	$statistics_endpoint = new WPSEO_Endpoint_Statistics( $statistics_service );
+	$statistics_endpoint->register();
 
+	if ( WPSEO_OnPage::is_active() ) {
```

Comment:
Wouldn't it be nicer to make the `register` method check for Ryte being active?

Reply:
This will avoid instanciating the Server (and loading the PHP file), thus saving memory and disk access in the process.